### PR TITLE
nuke target for verifying .NET SDK versions used

### PR DESCRIPTION
**Pull request by lachmatt**
## Why

Towards #3595 

## What

- Console application for verifying .NET SDK versions used in action workflows and dockerfiles.
- Nuke target running the app
- New workflow

## Tests

Local smoke tests.

## Checklist

~~- [ ] `CHANGELOG.md` is updated.~~
~~- [ ] Documentation is updated.~~
~~- [ ] New features are covered by tests.~~

### CODE REVIEW

File: tools/SdkVersionVerifier/VersionComparer.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+internal static class VersionComparer
+{
+    public static bool CompareVersions(DotnetSdkVersion expectedDotnetSdkVersion, DotnetSdkVersion version)
+    {
+        return CompareVersions(expectedDotnetSdkVersion, version.Net6SdkVersion, version.Net7SdkVersion, version.Net8SdkVersion);
+    }
+
+    public static bool CompareVersions(
+        DotnetSdkVersion expectedDotnetSdkVersion,
+        string? net6SdkVersion,
```
Comment (RassK) [inline tools/SdkVersionVerifier/VersionComparer.cs:15]: could be more dynamic, so new sdk versions will not cause issues?

File: tools/SdkVersionVerifier/ActionWorkflowVerifier.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        //   7.0.410
+        //   8.0.403
+
+        string? sdk6Version = null;
+        string? sdk7Version = null;
+        string? sdk8Version = null;
+
+        foreach (var version in dotnetVersionNode.ToString().Split())
+        {
+            if (version.StartsWith('6'))
```
Comment (RassK) [inline tools/SdkVersionVerifier/ActionWorkflowVerifier.cs:88]: for example group by first digit?

**Review by Kielek — APPROVED** [review]

File: tools/SdkVersionVerifier/ActionWorkflowVerifier.cs (hunk 2 above)
Comment (lachmatt) [reply tools/SdkVersionVerifier/ActionWorkflowVerifier.cs:88]: Do you mean to sort by the first digit?
That'd work as well. Note there are workflows where single .NET SDK version is used, we'd probably assume there that single version is of .NET SDK 8. I'm not sure if suggested approach is better.
I'd prefer to include modifications/improvements in a separate PR.

File: tools/SdkVersionVerifier/VersionComparer.cs (hunk 1 above)
Comment (lachmatt) [reply tools/SdkVersionVerifier/VersionComparer.cs:15]: I considered other approaches, but decided to start simple. I'd prefer to include improvements in a separate PR, unless they are urgent.

**Review by RassK — APPROVED** [review]

**Pull request merged**